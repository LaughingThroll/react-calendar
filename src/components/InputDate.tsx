import React from "react"

interface IInputDate extends React.InputHTMLAttributes<HTMLInputElement> {
  title?: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const InputDate: React.FC<IInputDate> = ({ title, ...rest }) => {
  return (
    <label className="input-date">
      {title} <input className="input-date__field" type="date" {...rest} />
    </label>
  )
}

export default InputDate
