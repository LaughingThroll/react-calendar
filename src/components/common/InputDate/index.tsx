import React from 'react'
import styles from './inputDate.module.scss'

export interface InputDateProps extends React.InputHTMLAttributes<HTMLInputElement> {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  label?: string
}

const InputDate: React.FC<InputDateProps> = ({ label, ...rest }) => {
  const { inputDate, inputDate__field } = styles
  return (
    <label className={inputDate}>
      {label} <input className={inputDate__field} type="date" {...rest} />
    </label>
  )
}

export default React.memo(InputDate)
