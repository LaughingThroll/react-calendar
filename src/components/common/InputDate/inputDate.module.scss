@import 'styles/utils/variables';

.inputDate {
  display: block;

  &__field {
    position: relative;
    border-bottom: 1px solid $input-date-border;
    font-size: 14px;
    line-height: 21px;
    font-weight: 600;
    width: 100%;
    height: 21px;
    outline: none;
    cursor: pointer;

    &::-webkit-calendar-picker-indicator {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-image: none;
      margin: 0;
      padding: 0;
      cursor: pointer;
    }
  }
}
