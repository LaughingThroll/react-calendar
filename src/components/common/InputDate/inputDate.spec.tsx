import React from 'react'
import { shallow } from 'enzyme'
import InputDate, { InputDateProps } from './'

const setUp = (props: InputDateProps) => shallow(<InputDate {...props} />)

describe('InputDate render', () => {
  it('default render', () => {
    const tree = setUp({ onChange: jest.fn() })

    expect(tree).toMatchSnapshot()
    expect(tree.find('label')).toHaveLength(1)
    expect(tree.find('input')).toHaveLength(1)
    expect(tree.find('input').prop('type')).toEqual('date')
  })
})

describe('InputDate props', () => {
  it('props label', () => {
    const tree = setUp({ label: 'Label', onChange: jest.fn() })

    expect(tree.find('label').text().includes('Label')).toBe(true)
  })

  it('props onChange', () => {
    const props = { onChange: jest.fn() }
    const tree = setUp(props)
    const $input = tree.find('input')

    $input.simulate('change')

    expect(props.onChange).toHaveBeenCalled()
  })
})
