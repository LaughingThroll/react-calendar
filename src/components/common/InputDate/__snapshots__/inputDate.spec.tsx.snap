// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDate render default render 1`] = `
<label
  className="inputDate"
>
   
  <input
    className="inputDate__field"
    onChange={[MockFunction]}
    type="date"
  />
</label>
`;
